backpack/extensions/firstorder/batch_l2_grad/rnn.py

```diff
@@ -0,0 +1,72 @@
+"""Contains BatchL2RNN."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Tuple
+
+from torch import Tensor
+from torch.nn import Module
+
+from backpack.core.derivatives.rnn import RNNDerivatives
+from backpack.extensions.firstorder.base import FirstOrderModuleExtension
+
+if TYPE_CHECKING:
+    from backpack.extensions import BatchL2Grad
```

Comment:
Just for clarification: When is `TYPE_CHECKING` turned on?

Reply:
During runtime it is never turned on. Only for type checkers.
[Reference](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)